pyEOS/eos.py

```diff
@@ -198,6 +210,29 @@ def replace_config(self, config=None, force=False):
         else:
             raise exceptions.CommandError(result[1]['messages'][0])
 
+    def merge_config(self, config=None):
+        """
+        Applies the configuration changes on the device. You can either commit the changes on the candidate_config
+        attribute or you can send the desired configuration as a string. Note that the current configuration of the
+        device is merged with the new configuration.
+
+        :param config: String containing the desired configuration. If set to None the candidate_config will be used
+
+        """
+        if config is None:
+            config = self.candidate_config.to_string()
+
+        commands = config.split('\n')
+        if 'configure' is not commands[0]:
+            commands.insert(0, 'configure')
+
+        result = self.run_commands([commands])
+
+        if 'Invalid' not in result[1]['messages'][0]:
```

Comment:
Shouldn´t we catch the error in the run_commands instead for consistency and reusability?